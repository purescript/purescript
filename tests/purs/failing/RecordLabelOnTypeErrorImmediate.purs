-- @shouldFailWith TypesDoNotUnify
module NestedRecordLabelOnTypeError where

record :: { a :: Int }
record = { a: "a" } -- Triggers an error, but the label is explicitly not added since it caused other errors to be worse. See https://github.com/purescript/purescript/pull/4411 for more information.
