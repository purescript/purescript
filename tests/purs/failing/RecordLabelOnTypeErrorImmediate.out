Error found:
in module [33mNestedRecordLabelOnTypeError[0m
at tests/purs/failing/RecordLabelOnTypeErrorImmediate.purs:5:15 - 5:18 (line 5, column 15 - line 5, column 18)

  Could not match type
  [33m        [0m
  [33m  String[0m
  [33m        [0m
  with type
  [33m     [0m
  [33m  Int[0m
  [33m     [0m

while checking that type [33mString[0m
  is at least as general as type [33mInt[0m
while checking that expression [33m"a"[0m
  has type [33mInt[0m
in value declaration [33mrecord[0m

See https://github.com/purescript/documentation/blob/master/errors/TypesDoNotUnify.md for more information,
or to contribute content related to this error.

